code/modules/mob/living/living_say.dm

```diff
@@ -212,36 +212,81 @@ GLOBAL_LIST_INIT(department_radio_keys, list(
 	return 1
 
 /mob/living/Hear(message, atom/movable/speaker, datum/language/message_language, raw_message, radio_freq, list/spans, list/message_mods = list())
-	SEND_SIGNAL(src, COMSIG_MOVABLE_HEAR, args)
-	if(!client)
-		return
+	if(HAS_TRAIT(speaker, TRAIT_SIGN_LANG)) //Checks if speaker is using sign language
+		SEND_SIGNAL(src, COMSIG_MOVABLE_HEAR, args)
+		if(!client)
+			return
+
+
+		var/deaf_message
+		var/deaf_type
+		if(speaker != src)
+			if(!radio_freq) //Overrides deafness check
+				deaf_message = compose_message(speaker, message_language, raw_message, radio_freq, spans, message_mods)
+				deaf_type = 1
+		else
+			deaf_message = compose_message(speaker, message_language, raw_message, radio_freq, spans, message_mods)
+			deaf_type = 2
+
+
+		if (client?.prefs.chat_on_map && !(stat == UNCONSCIOUS || stat == HARD_CRIT) && (client.prefs.see_chat_non_mob || ismob(speaker)))
+			create_chat_message(speaker, message_language, raw_message, spans)
+
+		if(HAS_TRAIT(src, TRAIT_BLIND))
+			return FALSE
+
+
+		message = compose_message(speaker, message_language, raw_message, radio_freq, spans, message_mods)
+
+		show_message(message, MSG_AUDIBLE, deaf_message, deaf_type, avoid_highlighting = speaker == src)
```

Comment:
I overlooked that MSG_AUDIBLE part of it, I should have realized that means there's a visual version I could use too. Making that change now.

Reply:
I just tested it, even with eyes taken out I could see the message. I'll leave it just in case it takes care of anything else I couldn't think of, but I'll leave in the is_blind check just to make sure it 100% works

Reply:
Aight that's ok